storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/ConsumerTask.java

```
@@ -120,6 +174,29 @@ public void run() {
         }
     }
 
+    private void syncCommittedDataAndOffsets(boolean forceSync) {
+        boolean noOffsetUpdates = committedPartitionToConsumedOffsets.equals(partitionToConsumedOffsets);
+        if (noOffsetUpdates || !forceSync && time.milliseconds() - lastSyncedTimeMs < committedOffsetSyncIntervalMs) {
+            log.debug("Skip syncing committed offsets, noOffsetUpdates: {}, forceSync: {}", noOffsetUpdates, forceSync);
+            return;
+        }
+
+        try {
+            // todo sync the snapshot file
+            for (TopicIdPartition topicIdPartition : assignedTopicPartitions) {
+                int metadataPartition = topicPartitioner.metadataPartition(topicIdPartition);
+                remotePartitionMetadataEventHandler.syncLogMetadataDataFile(topicIdPartition, metadataPartition,
+                                                                            partitionToConsumedOffsets.get(metadataPartition));
+            }
+
+            committedOffsetsFile.writeEntries(partitionToConsumedOffsets);
+            committedPartitionToConsumedOffsets = new HashMap<>(partitionToConsumedOffsets);
+            lastSyncedTimeMs = time.milliseconds();
+        } catch (IOException e) {
+            log.error("Error encountered while writing committed offsets to a local file", e);
```

Could we integrate this with logDirFailureChannel?

---

For now, we will log an error if there are any issues in writing to the file. There are no direct dependencies to `logDirFailureChannel` as it is part of RLMM implementation. We can address this in PRs when remote log subsystem is added.

---

This is an IOException from local storage. So, why do we want to wait when remote log subsystem is added? Ideally, we want to integrate it with logDirFailureChannel. If we want to do that later, it seems that we should at least throw an exception to stop the thread?